### DISCUSSION TITLE
granted registry add --ref option

### DISCUSSION THREAD

**Discussion by ohmer** [fwdcloudsec/granted#531] [Q&A] [answered]
Hi there,

I am looking to add a registry to test my changes currently sitting in a branch.
Found out a hidden `ref` flag in the code but it does not seem to be effective. 
Tried different reference types: long commit hash and branch name. Examples:
```
granted registry add --name test --url https://github.com/blah/blah.git --ref f916eee1eee0900c94cd5e2cb013f4ad858f6b04 -p path
granted registry add --name test --url https://github.com/blah/blah.git --ref branch -p path
```

Looking at my `$HOME/.granted/registries/test` and output, it fetches the default branch regardless passed `--ref` value.

Is it supposed to be working and I am missing something or maybe work in progress?

Using version `0.19.1`

Thanks!

**Comment by Eddie023** [comment, marked as answer]
Hi @ohmer, One of the maintainers from Granted here. You are correct, support for "ref" has not been implemented yet. 

Are you trying to test your changes without pushing to Git Hub? One way you can do that although not ideal is by creating a new registry that points to your local repo. You can follow the following steps: 

1.  create a new branch and commit your changes 
2. run "granted registry add -n test -u "<path_to_your_local_repo_directory>"  
3. Verify your changes and remove using "granted registry remove" 

Let me know if you have any more questions regarding this.

**Reply by ohmer** [reply]
Thanks for the tip, that's a work around and I will patiently wait for someone to contribute the `ref` implementation!
